packages/svelte/src/compiler/phases/1-parse/read/script.js

```diff
@@ -60,14 +42,56 @@ export function read_script(parser, start, attributes) {
 	// TODO is this necessary?
 	ast.start = script_start;
 
+	/** @type {'default' | 'module'} */
+	let context = 'default';
+
+	for (const attribute of /** @type {Attribute[]} */ (attributes)) {
+		if (RESERVED_ATTRIBUTES.includes(attribute.name)) {
+			e.script_reserved_attribute(attribute, attribute.name);
+		}
+
+		if (!ALLOWED_ATTRIBUTES.includes(attribute.name)) {
+			w.script_unknown_attribute(attribute);
+		}
+
+		if (attribute.name === 'module') {
+			if (attribute.value !== true) {
+				// Deliberately a generic code to future-proof for potential other attributes
+				e.script_invalid_attribute_value(attribute, attribute.name);
+			}
+
+			context = 'module';
+		}
+
+		if (attribute.name === 'context') {
+			if (attribute.value === true || !is_text_attribute(attribute)) {
+				throw new Error('TODO');
```

Comment:
Wasn't this supposed to be `e.script_invalid_context(attribute);`?

Reply:
yes! good spot, thanks — #12973